test/rekt/features/broker/feature.go

```diff
@@ -743,3 +898,35 @@ func brokerSubscriberLongMessage() *feature.Feature {
 	)
 	return f
 }
+
+type eventTestCase struct {
+	Type       string
+	Source     string
+	Extensions map[string]interface{}
+}
+
+// toEventMatcher converts the test case to the event matcher
+func (tc eventTestCase) toEventMatcher() test.EventMatcher {
+	var matchers []test.EventMatcher
+	if tc.Type == eventingv1.TriggerAnyFilter {
+		matchers = append(matchers, test.ContainsAttributes(spec.Type))
+	} else {
+		matchers = append(matchers, test.HasType(tc.Type))
+	}
+
+	if tc.Source == eventingv1.TriggerAnyFilter {
+		matchers = append(matchers, test.ContainsAttributes(spec.Source))
+	} else {
+		matchers = append(matchers, test.HasSource(tc.Source))
+	}
+
+	for k, v := range tc.Extensions {
+		if v == eventingv1.TriggerAnyFilter {
+			matchers = append(matchers, test.ContainsExtensions(k))
+		} else {
+			matchers = append(matchers, test.HasExtension(k, v))
+		}
+	}
+
+	return test.AllOf(matchers...)
+}
```

Comment:
Even though this looks less verbose I would make explicit for each test case and each sink their matcher in the `tests` table, so that by looking at the table I can clearly see:

given:
```golang
		eventsToSend []eventTestCase
		eventFilters [string]eventTestCase /* map[<sink-name>]eventTestCase */
```
I expect to match:
```golang
               matcherBySink map[string]test.EventMatcher /* map[<sink-name>]test.EventMatcher */
```

Reply:
ok , understood